src/components/ImageView/ImageView.js

```diff
@@ -574,7 +582,18 @@ export default observer(
           e.target === item.stageRef ||
           findClosestParent(
             e.target,
-            el => el.nodeType === 'Group' && ['ruler', 'segmentation'].indexOf(el?.attrs?.name) > -1,
+            el => {
+              if (el.nodeType === 'Group') {
+                if ('ruler' === el?.attrs?.name) {
+                  return true;
+                }
+                if ((!isFF(FF_DBLCLICK_DELAY) || !isMoveTool)
```

Comment:
please, rewrite this! extract lambda into separate named function, add comment, simplify condition